### PR TITLE
Add Twig responder

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: src/Component/Symfony/Request/State/TwigResponder.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+        $content = $this->twig->render(
+            $operation->getTemplate() ?? '',
+            $this->getTwigContext($data, $operation, $request),
+        );
+
+        return new Response($content);
+    }
+
+    private function getTwigContext(mixed $data, Operation $operation, Request $request): array
```
Comment (loic425) [inline src/Component/Symfony/Request/State/TwigResponder.php:74]: This twig context will be moved into a new service later which will be decorated for legacy variables.

File: src/Component/Symfony/Request/State/TwigResponder.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        /** @var FormInterface|null $form */
+        $form = $request->attributes->get('form');
+
+        if (null !== $form) {
+            $context['form'] = $form->createView();
+        }
+
+        if ($operation instanceof CollectionOperationInterface) {
+            $context['resources'] = $data;
+            $context[$operation->getResource()?->getPluralName() ?? ''] = $data;
```
Comment (loic425) [inline src/Component/Symfony/Request/State/TwigResponder.php:87]: the possibility of having a $context[''] = $data will be removed later too.

File: src/Component/Symfony/Routing/RedirectHandler.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        return new RedirectResponse($this->router->generate($route, $parameters));
+    }
+
+    private function parseResourceValues(array $parameters, mixed $data): array
+    {
+        $accessor = PropertyAccess::createPropertyAccessor();
+
+        if (empty($parameters)) {
+            if (\is_object($data) && $accessor->isReadable($data, 'id')) {
+                return ['id' => $accessor->getValue($data, 'id')];
```
Comment (loic425) [inline src/Component/Symfony/Routing/RedirectHandler.php:52]: I'd like to improve that part later. It comes from the current RedirectHandler.

File: src/Component/Symfony/Request/State/TwigResponder.php [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Twig\Environment;
+
+final class TwigResponder implements ResponderInterface
+{
+    public function __construct(
+        private RedirectHandler $redirectHandler,
+        private ?Environment $twig,
```
Comment (lchrusciel) [inline src/Component/Symfony/Request/State/TwigResponder.php:35]: Do we need to allow for null env?

**Review by lchrusciel — APPROVED** [review]

File: src/Component/Symfony/Request/State/TwigResponder.php (hunk 4 above)
Comment (loic425) [reply src/Component/Symfony/Request/State/TwigResponder.php:35]: yes Twig, we have smoke tests without Twig to place it on dev requirements later.

**Pull request merged**